### PR TITLE
feat: Auto register Third party devices on Dashboard #NEBD-2

### PR DESCRIPTION
**Pull request by pritamghanghas**
* automatically register third party devices on startup
* keep the registration state saved in sqlite db
* add buys loop env var that prevents diag startup for debugging

**[Issue](https://nebraltd.atlassian.net/browse/NEBD-2)**

- Link: https://nebraltd.atlassian.net/browse/NEBD-2
- Summary:

**How**

**Screenshots**

**References**

Closes: #470
Closes: #97
Closes: #371

**Checklist**

- [ ] Tests added
- [ ] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: hw_diag/utilities/dashboard_registration.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+    return device_payload
+
+
+def register_third_party_miner() -> None:
+    '''
+    tries to register a non nebra miner with dashboard
+    '''
+    try:
+        diagnostics = cached_diagnostics_data()
```
Comment (pritamghanghas) [inline hw_diag/utilities/dashboard_registration.py:57]: @kashifpk Can we remove this check. I think we get "400, already registered for nebra miners".  I don't really know the background of why this is a seaprate api.

**Review by MuratUrsavas — APPROVED** [review]
LGTM!

**Comment by shawaj** [comment]
Quick question - what does it display if the miner is already claimed in dashboard but the free trial has expired and they haven't paid?

@KevinWassermann94 maybe saying about claiming the free trial doesn't make sense in this case I'm not sure

File: hw_diag/utilities/dashboard_registration.py (hunk 1 above)
Comment (kashifpk) [reply hw_diag/utilities/dashboard_registration.py:57]: @pritamghanghas Haven't looked at this code before but from what I see in the code, this should just be about caching local diagnostics data and not affect dashboard registration. Better keep it in place as it's probably meant to reduce actual diagnostic runs.

**Pull request merged**